ngraph.mk

```diff
@@ -0,0 +1,175 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Including this .mk file has the following effects:
+#
+# - If any problem is noticed, calls $(error ...).
+#
+# - Assuming no error is noticed:
+#   - Sets the following variables (perhaps to the empty string):
+#     NGRAPH_CFLAGS  - Compiler args needed to build the MXnet-nGraph bridge code.
+#     NGRAPH_LDFLAGS - Linker args needed to build the MXnet-nGraph bridge code.
+#
+#   - In whatever way is appropriate, defines new goals and adds them as
+#     dependencies of the 'all' and/or 'clean' targets.
+#
+# Input variables:
+#   USE_NGRAPH
+#   USE_NGRAPH_DISTRIBUTED
+#   USE_NGRAPH_IE
+#   ADD_NGRAPH_LIBDIR_TO_MXNET_RPATH
+#   NGRAPH_DIR
+#   NGRAPH_EXTRA_CMAKE_FLAGS
+#   NGRAPH_EXTRA_MAKE_FLAGS
+
+#===================================================================================================
+# Check for some configuration problems...
+#===================================================================================================
+
+ifeq ($(USE_NGRAPH), 1)
+  ifeq ($(USE_MKLDNN), 1)
+    $(error "Cannot have both USE_NGRAPH=1 and USE_MKLDNN=1: they require different MKLDNN versions.")
+  endif
+endif
+
+#===================================================================================================
+
+ifeq ($(USE_NGRAPH), 1)
+  ifeq ($(NGRAPH_DIR),)
+    NGRAPH_INSTALL_DIR=$(ROOTDIR)/3rdparty/ngraph.install
```

Comment:
ngraph.install or ngraph/install?

Reply:
Could go either way.  In general I try to avoid mixing source, build, and install directory trees.  I can change this if you feel strongly.

Reply:
Nope, just wanted to make sure it was intentional.